### DISCUSSION TITLE
How to use System.Linq.Enumerable methods?

### DISCUSSION THREAD

**Discussion by thinhbuzz** [vfsfitvnm/frida-il2cpp-bridge#327] [Q&A]
I have tried several ways but can't figure out how to call these functions, thanks for the review
```typescript
const SystemInt32 = getMSCorlibClass('System.Int32');
const List = getMSCorlibClass('System.Collections.Generic.List`1');
const list = List.inflate(SystemInt32).new();
list.method('.ctor', 0).invoke();
list.method('Add').invoke(5);
list.method('Add').invoke(6);
list.method('Add').invoke(7);
console.log(list);
console.log(Enumerable.method('First', 1).inflate(SystemInt32));
console.log(Enumerable.method('First', 1).inflate(SystemInt32).invoke(list));
```
output
```bash
System.Collections.Generic.List`1[System.Int32]
static System.Int32 First(System.Collections.Generic.IEnumerable<System.Int32> source);
Error: access violation accessing 0x0
# in some cases i will get error
Error: abort was called
```
[System.Linq.Enumerable](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=net-7.0)

**Comment by vfsfitvnm** [comment]
`First<System.Int32>` doesn't exist in your assembly. As you can see, there's no relative virtual address when printing the inflated method, and its `virtualAddress` is `NULL` (hence *access violation accessing 0x0*).

However, I should probably catch that exception so you immediately know why.

Are you on Windows by any chance? I have to make sure Frida returns the exact same error message

**Reply by thinhbuzz** [reply]
i have windows device but never tried with windows app.
anyway, thank you for your information, I will check and provide more information

**Reply by vfsfitvnm** [reply]
```ts
new NativeFunction(NULL, "void", [])();
```
is enough to verify
